templates/store/search.html

```diff
@@ -94,8 +94,10 @@ <h2>We couldn't find the category <div class="no-search-query">{{category_displa
         {% if links.prev %}<a href="{{ links.prev }}">&#8249;&nbsp;Previous</a>{% endif %}
       </div>
       <div class="u-float-right u-align--right">
-        {% if links.next %}<a href="{{ links.next }}">Next&nbsp;&#8250;</a>{% endif %}
-        {% if links.last %}&nbsp;<a href="{{ links.last }}">Last&nbsp;&#187;</a>{% endif %}
+        {% if total > 47 %}
```

Comment:
Why does the total need to be over 47? There should be a comment explaining the number, or better yet this should be a variable.
This condition would also be better in the related `view.py` - if the total isn't over 47 `links.next` and `links.last` don't get set and therefore won't be shown in the template.
Adding the condition against the magic number in the template means that there is now 4 potential conditions:
not over 47, next and last links don't show
over 47 and next link isn't defined so doesn't show, but last link is defined and shows
over 47 and next link is defined so shows, but last link isn't defined so doesn't show
over 47 and next and last links are both undefined so don't show
These potential 4 conditions are also spread between the template and the `views.py` which means if there's an error it's harder to debug which condition is failing.

Reply:
The problem is that currently for category pages we show 47 snaps on the first page and 44 on the rest of the pages. Because of this we show snaps 45, 46 & 47 on both first and second page. After discussing with Robin and filing an [issue](https://github.com/canonical-web-and-design/snapcraft.io/issues/2457) the easiest/best thing to do is to update the design of the first page to show 44 snaps (like the other pages). 
Regarding the condition in the code, I will  add it to `views.py` as it would be easier to manage.  Basically it is needed to hide the "Last page" navigation if we have less than 47 snaps, as the last page would be page 2 and would contain maximum 3 snaps, already shown on page 1.

Reply:
I don't think we need to make that change as a second issue. 
In `views.py` you could change `19` to `16` on line 212, remove lines 148-157 (so there's no conditional changing things on category pages).
After that you should be able to remove the conditional added to the template too.